lib/fizzy/leb128.hpp

```diff
@@ -2,8 +2,18 @@
 
 namespace fizzy
 {
-inline constexpr uint64_t leb128_decode(const uint8_t*) noexcept
+inline constexpr uint64_t leb128u_decode(const uint8_t* input) noexcept
 {
-    return 0;
+    uint64_t result = 0;
+    uint8_t result_shift = 0;
```

Comment:
You should avoid using unsigned types smaller than native word size (`size_t`), especially in loops. The unsigned types guarantee wrapping behavior and compiler might need to emulate that with additional instructions. On the opposite, when using signed type the overflow is undefined behavior and compiler can assume the overflow never happens. Moreover, the true overflow can be detected with Undefined Behavior Sanitizer (very useful in fuzzing).

In such case I use `int` type indicating some signed integer value.

By changing type to `int`, we save one instruction in the loop: https://godbolt.org/z/xo7rZY.